apps/openmw/mwrender/npcanimation.cpp

```diff
@@ -1096,37 +1109,71 @@ const std::vector<const ESM::BodyPart *>& NpcAnimation::getBodyParts(const std::
+                                    bodypart.mData.mPart == ESM::BodyPart::MP_Wrist ||
+                                    bodypart.mData.mPart == ESM::BodyPart::MP_Forearm ||
+                                    bodypart.mData.mPart == ESM::BodyPart::MP_Upperarm;
+
+            bool isSameGender = isFemalePart(&bodypart) == female;
+
+            /* A fallback for the arms if 1st person is missing:
+             1. Try to use 3d person skin for same gender
+             2. Try to use 1d person skin for opposite gender
+             3. Try to use 3d person skin for opposite gender
+            */
+            if(firstPerson && isHand)
             {
-                if(firstPerson && (bodypart.mData.mPart == ESM::BodyPart::MP_Hand ||
-                                                   bodypart.mData.mPart == ESM::BodyPart::MP_Wrist ||
-                                                   bodypart.mData.mPart == ESM::BodyPart::MP_Forearm ||
-                                                   bodypart.mData.mPart == ESM::BodyPart::MP_Upperarm))
+                if(partFirstPerson != (firstPerson))
                 {
-                    /* Allow 3rd person skins as a fallback for the arms if 1st person is missing. */
+                    /* Allow 3rd person skins of same gender as a fallback for the arms if 1st person is missing. */
                     BodyPartMapType::const_iterator bIt = sBodyPartMap.lower_bound(BodyPartMapType::key_type(bodypart.mData.mPart));
                     while(bIt != sBodyPartMap.end() && bIt->first == bodypart.mData.mPart)
                     {
+                        // If we have substitution from p.2 or p.3 and better substitution from p.1 found
+                        if(parts[bIt->second] && isSameGender && isFemalePart(parts[bIt->second]) != female)
+                            parts[bIt->second] = &*it;
+
                         if(!parts[bIt->second])
                             parts[bIt->second] = &*it;
+
+                        ++bIt;
+                    }
+
+                    continue;
+                }
+                else
+                {
+                    BodyPartMapType::const_iterator bIt = sBodyPartMap.lower_bound(BodyPartMapType::key_type(bodypart.mData.mPart));
+                    while(bIt != sBodyPartMap.end() && bIt->first == bodypart.mData.mPart)
+                    {
+                        // If we have substitution from p.3 and better substitution from p.2 found
+                        if(parts[bIt->second] && !isSameGender && isFemalePart(parts[bIt->second]) != female)
+                            parts[bIt->second] = &*it;
+
+                        if(!parts[bIt->second])
+                            parts[bIt->second] = &*it;
+
                         ++bIt;
                     }
                 }
-                continue;
             }
 
+            // We do not need such fallback for non-hand bodyparts
```

Comment:
Hmm... It seems this comment is misleading.
I meaned that if a current bodypart is a hand, it is handled by fallback logic above. Otherwise we should skip this bodypart in this check.